drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -372,16 +410,91 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 	}
 
 	if (i == clk_values) {
-		dev_err(bus->dev, "%s: could not find clock value for bandwidth %d\n",
+multilane:
+		dev_dbg(bus->dev, "%s: could not find clock value for bandwidth %d, checking multi-lane\n",
 			__func__, bus->params.bandwidth);
-		return -EINVAL;
+
+		m_rt = list_last_entry(&bus->m_rt_list, struct sdw_master_runtime, bus_node);
+		/*
+		 * Find available manager lanes that connected to the first Peripheral. No need
+		 * to check all Peripherals available lanes because we can't use multi-lane
+		 * if we can't find any available lane for the first Peripheral.
+		 */
+		s_rt = list_first_entry(&m_rt->slave_rt_list, struct sdw_slave_runtime, m_rt_node);
+		slave_prop = &s_rt->slave->prop;
+
+		for (i = 1; i < SDW_MAX_LANES; i++) {
```

Comment:
this for {} is inside an if (i == clk_values). Do we want to use i again?

Reply:
> this for {} is inside an if (i == clk_values). Do we want to use i again?

It should be fine, we will not test `if (i == clk_values)` later. But, yeah, it will be more readable to use another variable. Updated